boards/BMS/Src/syscalls.c

```diff
@@ -113,7 +113,7 @@ __attribute__((weak)) int _write(int file, char *ptr, int len)
     return len;
 }
 
-caddr_t _sbrk(int incr)
+__attribute__((weak)) caddr_t _sbrk(int incr)
```

Comment:
The default implementation provided by STM32 is broken for multithreaded systems. Leave it as weak so it can be overridden by the patch.